administrator/components/com_media/views/media/view.html.php

```diff
@@ -85,4 +108,28 @@ protected function prepareToolbar()
 
 		JToolbarHelper::help('JHELP_CONTENT_MEDIA_MANAGER');
 	}
+
+	/**
+	 * Obtain list of supported providers
+	 *
+	 * @return string
+	 *
+	 * @since __DEPLOY_VERSION__
+	 */
+	public function getProviders()
```

Comment:
This function needs to be moved to the model. Then you can do in this class `$this->providers = $this->get('Providers');`